cmd/juju/application/bundle/bundle.go

```diff
@@ -350,59 +345,74 @@ func verifyBundle(data *charm.BundleData, bundleDir string, verifyConstraints fu
 		for _, err := range verr.Errors {
 			errs = append(errs, err.Error())
 		}
+	}
+
+	if len(errs) > 0 {
 		return errors.New("the provided bundle has the following errors:\n" + strings.Join(errs, "\n"))
 	}
-	return errors.Trace(verifyError)
+	return nil
 }
 
-func verifyMixedSeriesBasesMatch(data *charm.BundleData) error {
-	if data == nil {
-		return nil
+type bundleSeriesData struct {
+	Series string `yaml:"series,omitempty"`
+
+	Applications map[string]struct {
+		Series string `yaml:"series"`
+	} `yaml:"applications"`
+
+	Machines map[string]struct {
+		Series string `yaml:"series"`
+	} `yaml:"machines"`
+}
+
+func verifyNoSeries(dslist []charm.BundleDataSource) []string {
+	errs := []string{}
+
+	// The first dslist is always the bundle itself
+	bundleErrs := verifyBundleNoSeries(dslist[0].RawBundle())
```

Comment:
I know that there is a check above where this is called to check that dslist is empty, but I don't like this coding style. Instead, can this return an empty string if the list is empty.

Reply:
You mean like a catch guard? Sure thing